File: src/main/java/co/herod/adbwrapper/PropHelper.java
package co.herod.adbwrapper;

import java.util.Map;

@SuppressWarnings("WeakerAccess")
class PropHelper {

    static boolean hasPositiveValue(Map.Entry<String, String> entry) {
        return hasPositiveValue(entry.getValue());
    }

    static boolean hasPositiveValue(String value) {

        value = value.trim().toLowerCase();

        return value.equals("on") || value.equals("1") || Boolean.parseBoolean(value);
    }

    static boolean isValidProperty(Map.Entry<String, String> a) {
        return !a.getKey().contains(" ");
    }

    static boolean isKey(Map.Entry<String, String> entry, String key) {
        return entry.getKey().equals(key);
    }
}

File: adb-kotlin/src/main/kotlin/co/herod/adbwrapper/util/PropHelper.kt
package co.herod.adbwrapper.util

import co.herod.adbwrapper.model.DumpsysEntry

fun DumpsysEntry.hasPositiveValue(): Boolean = value.sanitised().hasPositiveValue()

fun hasPositiveValue(entry: Map.Entry<String, String>): Boolean = entry.value.sanitised().hasPositiveValue()

private fun String.sanitised(): String = trim().toLowerCase()

private fun String.hasPositiveValue(): Boolean = this == "on" || this == "1" || "true" in this
